airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/state/GlobalAsyncStateManager.java

```diff
@@ -196,6 +216,9 @@ private Long getStateIdAndIncrement(final StreamDescriptor streamDescriptor, fin
     }
     final Long stateId = descToStateIdQ.get(resolvedDescriptor).peekLast();
     final var update = stateIdToCounter.get(stateId).addAndGet(increment);
+    if (increment >= 0) {
```

Comment:
is `increment` ever negative?

Reply:
I don't think this is possible since it is called with a 0 value. Unfortunatelly, there is no ulong in java

Reply:
No it wont in the current flow. This is just a safety check that i put in place